File: python/array/01.two_sum.py

# Link --> https://leetcode.com/problems/two-sum/

def two_sum(nums, target):
    nums_dict = {}
    n = len(nums)

    for i in range(n):
        difference = target - nums[i]
        if difference in nums_dict:
            return [nums_dict[difference], i]
        nums_dict[nums[i]] = i

nums = [2,3,5,6]
target = 7

print(two_sum(nums, target))

File: java/array/01.TwoSum.java

// Link --> https://leetcode.com/problems/two-sum/

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int result = 0;
        for (int i = 0; i < nums.length; i++) {
            result = target - nums[i];
            for (int j = i + 1; j < nums.length; j++) {
                if (result == nums[j]) {
                    return new int[] {i, j};
                }
            }
        }
        return null;
    }
}